apps/meteor/client/views/room/composer/messageBox/RichTextMessageBox.tsx

```diff
@@ -0,0 +1,563 @@
+
+	const keyboardEventHandler = useEffectEvent((event: KeyboardEvent) => {
+		const { which: keyCode } = event;
+
+		const input = event.target as HTMLDivElement;
+
+		const isSubmitKey = keyCode === keyCodes.CARRIAGE_RETURN || keyCode === keyCodes.NEW_LINE;
+
+		if (isSubmitKey) {
+			const withModifier = event.shiftKey || event.ctrlKey || event.altKey || event.metaKey;
+			const isSending = (sendOnEnter && !withModifier) || (!sendOnEnter && withModifier);
+
+			event.preventDefault();
+			if (!isSending) {
+				chat.composer?.insertNewLine();
+				return false;
+			}
+			handleSendMessage();
+			return false;
+		}
+
+		if (chat.composer && handleFormattingShortcut(event, [...formattingButtons], chat.composer)) {
+			return;
+		}
+
+		if (event.shiftKey || event.ctrlKey || event.metaKey) {
+			return;
+		}
+
+		switch (event.key) {
+			case 'Escape': {
+				closeEditing(event);
+				if (!input.innerText.trim()) onEscape?.();
+				return;
+			}
+
+			case 'ArrowUp': {
+				const { selectionEnd } = getSelectionRange(input);
+
+				if (selectionEnd === 0) {
+					event.preventDefault();
+					event.stopPropagation();
+
+					onNavigateToPreviousMessage?.();
+
+					if (event.altKey) {
+						setSelectionRange(input, 0, 0);
+					}
+				}
+
+				return;
+			}
+
+			case 'ArrowDown': {
+				const { selectionEnd } = getSelectionRange(input);
+
+				if (selectionEnd === input.innerText.length) {
+					event.preventDefault();
+					event.stopPropagation();
+
+					onNavigateToNextMessage?.();
+
+					if (event.altKey) {
+						setSelectionRange(input, input.innerText.length, input.innerText.length);
+					}
+				}
+			}
+		}
+
+		onTyping?.();
+	});
+
+	const isEditing = useSyncExternalStore(chat.composer?.editing.subscribe ?? emptySubscribe, chat.composer?.editing.get ?? getEmptyFalse);
+
+	const isRecordingAudio = useSyncExternalStore(
+		chat.composer?.recording.subscribe ?? emptySubscribe,
+		chat.composer?.recording.get ?? getEmptyFalse,
+	);
+
+	const isMicrophoneDenied = useSyncExternalStore(
+		chat.composer?.isMicrophoneDenied.subscribe ?? emptySubscribe,
+		chat.composer?.isMicrophoneDenied.get ?? getEmptyFalse,
+	);
+
+	const isRecordingVideo = useSyncExternalStore(
+		chat.composer?.recordingVideo.subscribe ?? emptySubscribe,
+		chat.composer?.recordingVideo.get ?? getEmptyFalse,
+	);
+
+	const formatters = useSyncExternalStore(
+		chat.composer?.formatters.subscribe ?? emptySubscribe,
+		chat.composer?.formatters.get ?? getEmptyArray,
+	);
+
+	const isRecording = isRecordingAudio || isRecordingVideo;
+
+	const canSend = useReactiveValue(useCallback(() => roomCoordinator.verifyCanSendMessage(room._id), [room._id]));
+
+	/* const sizes = useContentBoxSize(textareaRef); */
+
+	const newSizes = useContentBoxSize(contentEditableRef);
+
+	const format = useFormatDateAndTime();
+
+	const joinMutation = useMutation({
+		mutationFn: async () => onJoin?.(),
+	});
+
+	const handlePaste = useEffectEvent((event: ClipboardEvent<HTMLDivElement>) => {
```

Comment:
We might need to strip the pasted text from styles, as it is today, some styles from the origin of the paste carry over

<img width="1554" height="137" alt="Screenshot 2025-07-14 at 10 28 57" src="https://github.com/user-attachments/assets/033f4fae-4c6d-4f48-80fa-5f86e6dad08d" />

<img width="1558" height="222" alt="Screenshot 2025-07-14 at 10 40 07" src="https://github.com/user-attachments/assets/9ea51ad0-2376-4f2a-8cc8-21bca94ebe67" />

This was not necessary before, since the MessageBox only supported text, with rich content that is an issue. 

I've tried some rich text that had some xss exploits in it and it didn't work, so we know the content is being sanitized at least